### PR TITLE
`Development`: Document exam exercise update notifications

### PR DESCRIPTION
**Pull request by tobias-lippert**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

The exam exercise update notification feature lacks documentation

### Description

Added documentation including screenshots.

### Steps for Testing
Have a look at the docs
Code Review
### Review Progress

Check the docs and the added information to the feature overview (shown in screencasts below)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Screenshots

https://github.com/ls1intum/Artemis/assets/84102468/f4037e2a-bd1b-4d05-b194-3235735ca441

### CODE REVIEW

**Review by MaximilianAnzinger — DISMISSED** [review]
LGTM one suggestion: Maybe it should be more obvious that manual merging is not possible in the online editor? Some highlighting since this might be a problem?

File: docs/user/exams/students_guide.rst [hunk 1]
```diff
@@ -277,6 +277,23 @@ Participating in Programming Exercises
     You can use the |resolve_conflict| button, which is then displayed instead of the submit button, to resolve the conflict within the online code editor.
     This will reset your changes to the latest commit. Manual merging is not possible with the online code editor.
 
+.. _updated_problem_statement:
+
+Updated Problem Statement during the Exam
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If your instructor updates the problem statement for an exercise during the exam, you will receive a notification as seen in the screenshot below and can toggle between a diff view and the new problem statement as shown in the second and third screenshot.
```
Comment (florian-glombik) [inline docs/user/exams/students_guide.rst:284]: I think that the sentence is missing some commas; I would also rather speak of switching between the dif and not toggling

```suggestion
If your instructor updates the problem statement for an exercise during the exam, you will receive a notification, as seen in the screenshot below, and can switch between a diff view and the new problem statement, as shown in the second and third screenshot.
```

**Review by florian-glombik — CHANGES_REQUESTED** [review]
Some comments on the formulations, the error with the headline must be fixed to ensure that the documentation compiles

File: docs/user/exams/instructors_guide.rst [hunk 2]
```diff
@@ -347,7 +347,16 @@ During the exam creation and configuration, you can create your exam and configu
 -----------------
 The exam conduction starts when the exam becomes visible to the students and ends when the latest working time is over. When the exam conduction begins, you cannot make any changes anymore to the :ref:`exam configuration <exam_creation_and_configuration>` or individual :ref:`student exams <student_exams>`. When the conduction starts, the students can access and start their exam. They can submit their solutions to the exercises within the given individual working time. When a student submits the exam, they cannot make any changes anymore to his exercise submissions. For more information, see :ref:`participating in the online exam <participation_guide>`.
 
-    .. _exam_assessment:
+Updating an Exercise during the Exam
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
[the remark this answers is not in the document]
Comment (florian-glombik) [reply docs/user/exams/instructors_guide.rst:351]: Suggestion on wrong part of the code

File: docs/user/exams/students_guide.rst [hunk 3]
```diff
@@ -277,6 +277,23 @@ Participating in Programming Exercises
     You can use the |resolve_conflict| button, which is then displayed instead of the submit button, to resolve the conflict within the online code editor.
     This will reset your changes to the latest commit. Manual merging is not possible with the online code editor.
 
+.. _updated_problem_statement:
+
+Updated Problem Statement during the Exam
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (florian-glombik) [inline docs/user/exams/students_guide.rst:283]: The docu does not compile without this change

```suggestion
Updated Problem Statement during the Exam
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: docs/user/exams/instructors_guide.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (florian-glombik) [reply docs/user/exams/instructors_guide.rst:351]: See https://github.com/ls1intum/Artemis/pull/7294#discussion_r1341288830 for the intended change

**Review by laurenzfb — DISMISSED** [review]
Overall LGTM but I do agree on the suggested changes by @florian-glombik

**Review by aplr — DISMISSED** [review]
LGTM w/ change requests from @florian-glombik also. Good to read & understandable otherwise.

**Review by b-fein — DISMISSED** [review]

**Review by laurenzfb — APPROVED** [review]
LGTM after changes!

**Review by b-fein — APPROVED** [review]

**Review by MaximilianAnzinger — APPROVED** [review]

**Review by aplr — APPROVED** [review]

**Review by florian-glombik — DISMISSED** [review]
Thanks for incorporating the feedback! Looks good to me now!

**Review by b-fein — DISMISSED** [review]

**Review by aplr — APPROVED** [review]

**Review by laurenzfb — DISMISSED** [review]

**Review by krusche — CHANGES_REQUESTED** [review]
Could you also document this feature on https://artemis.in.tum.de/features/instructors and on https://artemis.in.tum.de/features/students ?

And does it make sense to put the exam announcement and time extension in this PR as well?

**Comment by tobias-lippert** [comment]
@krusche Gonna add that. Regarding the second aspect, I do not know much about the exam announcement feature but as far as I know they work still a bit differently at the moment. @aplr What do you think? Do you want to add your documenation to this PR?

**Comment by aplr** [comment]
@krusche @tobias-lippert If you don't have opposing views, I'd prefer to add time extensions & announcements in another PR.

**Comment by krusche** [comment]
> @krusche @tobias-lippert If you don't have opposing views, I'd prefer to add time extensions & announcements in another PR.

As long as we document this asap, I'm fine

**Review by b-fein — APPROVED** [review]

**Review by laurenzfb — APPROVED** [review]

**Review by MaximilianAnzinger — APPROVED** [review]

**Pull request merged**